stl/inc/xlocnum

```diff
@@ -1136,26 +1136,19 @@ __PURE_APPDOMAIN_GLOBAL locale::id num_get<_Elem, _InIt>::id;
 #pragma clang diagnostic pop
 #endif // __clang__
 
-template <class _Ty>
-struct _Hex_float_precision;
-
-template <>
-struct _Hex_float_precision<double> {
-    // the number of hexits needed to represent (DBL_MANT_DIG - 1) bits after the radix point exactly
-    static constexpr int value = ((DBL_MANT_DIG - 1) + 3) / 4;
-};
-
-template <>
-struct _Hex_float_precision<long double> {
-    // the number of hexits needed to represent (LDBL_MANT_DIG - 1) bits after the radix point exactly
-    static constexpr int value = ((LDBL_MANT_DIG - 1) + 3) / 4;
-};
-
 template <class _Ty>
 int _Float_put_desired_precision(const streamsize _Precision, const ios_base::fmtflags _Float_flags) {
     const bool _Is_hex = _Float_flags == (ios_base::fixed | ios_base::scientific);
     if (_Is_hex) {
-        return _Hex_float_precision<_Ty>::value;
+        if constexpr (is_same_v<_Ty, double>) {
+            // the number of hexits needed to represent (DBL_MANT_DIG - 1) bits after the radix point exactly

```

Comment:
Could we combine this comment and line 1147 into a single comment describing the value this function returns on line 1141?

Reply:
This one's a bit complicated because these comments describe the hexfloat codepath, not the function as a whole.

I'll fuse the comments and then add a second comment for the function as a whole.